src/components/AssetImporter/AssetImporter.tsx

```diff
@@ -0,0 +1,257 @@
+            blob
+          }
+        })
+    )
+
+    const contents = files.reduce<Record<string, Blob>>((contents, file) => {
+      contents[file.name] = file.blob
+      return contents
+    }, {})
+
+    const id = getSHA256(`${assetPackId}/${file.name}`)
+
+    return {
+      id,
+      fileName: file.name,
+      asset: {
+        id,
+        assetPackId,
+        name: manifestParsed ? manifestParsed.name : cleanAssetName(file.name),
+        tags: manifestParsed ? manifestParsed.tags : [],
+        category: manifestParsed ? manifestParsed.category : 'decorations',
+        url: assetModel,
+        contents,
+        metadata: getDefaultMetadata()
+      }
+    } as ImportedFile
+  }
+
+  handleModelFile = (file: File) => {
+    const { assetPackId } = this.state
+    const id = getSHA256(`${assetPackId}/${file.name}`)
+    return createDefaultImportedFile(id, assetPackId, file)
+  }
+
+  handleDropAccepted = async (acceptedFiles: File[]) => {
+    const { files } = this.state
+    let newFiles: Record<string, ImportedFile> = {}
+
+    for (let file of acceptedFiles) {
+      let outFile: ImportedFile | null = null
+      const extension = getExtension(file.name)
+
+      try {
+        if (!extension) {
+          throw new Error(t('asset_pack.import.invalid'))
+        }
+
+        if (extension === '.zip') {
+          outFile = await this.handleZipFile(file)
+        } else if (extension === '.gltf' || extension === '.glb') {
+          outFile = this.handleModelFile(file)
+        }
+
+        if (outFile) {
+          let mappings = rawMappingsToObjectURL(outFile.asset.contents)
+          const { image, info } = await getModelData(mappings[outFile.asset.url], {
+            mappings,
+            thumbnailType: outFile.asset.category === GROUND_CATEGORY ? '2d' : '3d'
+          })
+          revokeMappingsObjectURL(mappings)
+
+          outFile.asset.thumbnail = image
+          outFile.asset.metadata.metrics = info
+        }
+      } catch (e) {
+        // TODO: analytics
+        outFile = {
+          id: getSHA256(file.name),
+          fileName: file.name,
+          isCorrupted: true
+        } as ImportedFile
+      }
+
+      if (outFile) {
+        newFiles[outFile!.id] = outFile
+      }
+    }
+
+    const fileRecord = { ...files, ...newFiles }
+    this.setState({ files: fileRecord })
+  }
+
+  handleDropRejected = (rejectedFiles: File[]) => {
+    console.log('rejected', rejectedFiles)
+  }
+
+  handleRemoveProject = (id: string) => {
+    const { [id]: _, ...files } = this.state.files
+    this.setState({ files })
+  }
+
+  handleSubmit = () => {
+    const { assetPackId, files } = this.state
+    const assets = Object.values(files).map(file => file.asset)
+
+    this.props.onSubmit({
+      id: assetPackId,
+      title: '',
+      thumbnail: '',
+      url: '',
+      isLoaded: false,
+      assets
+    })
+  }
+
+  render() {
+    const { files } = this.state
+    const fileKeys = Object.keys(files)
```

Comment:
Here we are doing an `Object.keys` and then an `Object.values` below (for the `items` prop). What about just using the `items` and drop the `Object.keys`? Like:

```tsx
    const items = Object.values(files)
    const buttonText = items.length > 1 ? t('asset_pack.import.action_many', { count: items.length }) : t('asset_pack.import.action')
    const hasCorrupted = items.some(item => !!item.isCorrupted)
    const canImport = items.length > 0 && !hasCorrupted
```

And then just pass `items` to `<FileImport>`

Reply:
👌